pkg/features/kube_features.go

```diff
@@ -554,6 +554,12 @@ const (
 	// e.g. emptyDir:
 	//        medium: HugePages-1Gi
 	HugePageStorageMediumSize featuregate.Feature = "HugePageStorageMediumSize"
+
+	// owner: @bswartz
+	// alpha: v1.18
+	//
+	// Enables usage of any object for volume data source in PVCs
+	GenericPVCDataSource featuregate.Feature = "GenericPVCDataSource"
```

Comment:
I'm open to suggestions on the specific feature gate name.

I know what you mean about this feature gate implicitly enabling the other feature gates, but I'm not sure how that works at a technical level. Currently, if you enable this feature gate, it doesn't matter whether the other gates are enabled or not because the behavior of the new one overrides them (see unit tests that prove this to be true). However I'm not sure what more we can do to make this clear. Should we generate an error if conflicting feature gates are set?

Reply:
> Currently, if you enable this feature gate, it doesn't matter whether the other gates are enabled or not because the behavior of the new one overrides them

That's exactly what we DO want IMO, I didn't see the unit tests that actually test that path though (eg snapshots work when VolumeSnapshot feature is disabled but Generic is enabled).  Looking again I see how you handled that, it just wasn't explicitly clear the first time I looked at it.

Reply:
VolumeSnapshotDataSource remains beta in 1.18.  By enabling this generic feature gate, it implies VolumeSnapshotDataSource is always enabled?  I think we still need to check VolumeSnapshotDataSource feature gate even if this new generic feature gate is enabled.

Reply:
Discussed with @thockin. The behavior of the new generic datasource should supercede the other two feature gates. So I think the way it is implemented is fine. Agree with @j-griffith that there could be more test cases explicitly testing this.

Reply:
Will this affect our E2E tests?

Reply:
@msau42 I did cover those specific cases. I think @j-griffith agrees. Can you describe the combination that's not tested?

Reply:
@bswartz Actually I was in favor of testing all permutations :)  That being said, you cover the important ones here I think; and hopefully https://github.com/kubernetes/kubernetes/pull/88686 will land and make half of this irrelevant.

That being said:
1.  I still think the naming is horrible and will cause confusion
2.  You need to update the docstrings in the types files
3.  Additional tests are needed, but they can be added later as this is alpha proposal IMO

Reply:
Okay I added more permutations. Is "AnyVolumeDataSource" the feature gate string you prefer? I'm okay with that one if there are no objections.

Reply:
that's fine with me, I just want to use the same naming pattern / convention and leave PVC out of it (GenericPVCDataSource) is too close to "VolumePVCDataSource" and doesn't make sense against things like "VolumeSnapshotDataSource".  I know, it seems silly but I think it matters.